roadway_objects/test/roadway_objects_regular_map_launch_test.py

```diff
@@ -0,0 +1,310 @@
+    def __init__(self, sized_object: Sized) -> None:
+        self.sized_object = sized_object
+        self.original_size = len(sized_object)
+
+    def __call__(self) -> bool:
+        return len(self.sized_object) > self.original_size
+
+
+@pytest.mark.launch_test
+def generate_test_description():
+    node_under_test = Node(
+        package="roadway_objects",
+        executable="roadway_objects_node",
+        name="node_under_test",
+        parameters=[{"vehicle_participant_type": "vehicle:car"}],
+    )
+
+    lanelet2_map_path = (
+        Path(get_package_share_directory("roadway_objects"))
+        / "test/data/town01_vector_map_1.osm"
+    )
+
+    lanelet2_map_loader = Node(
+        package="map_file_ros2",
+        executable="lanelet2_map_loader_exec",
+        name="lanelet2_map_loader",
+        parameters=[{"lanelet2_filename": str(lanelet2_map_path)}],
+        remappings=[("lanelet_map_bin", "semantic_map")],
+    )
+
+    launch_description = LaunchDescription(
+        [
+            node_under_test,
+            lanelet2_map_loader,
+            TimerAction(period=1.0, actions=[ReadyToTest()]),
+        ]
+    )
+
+    # These will get passed to the unittest test cases as keyword args
+    context = {"node_under_test": node_under_test}
+
+    return launch_description, context
+
+
+class TestRegularMap(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+
+        cls.context = Context()
+
+        rclpy.init(context=cls.context)
+        cls.test_harness_node = TestHarnessNode(context=cls.context)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        super().tearDownClass()
+
+        rclpy.shutdown(context=cls.context)
+
+    def test_external_object_list_regular_map(self):
+        trans_id = Transition.TRANSITION_CONFIGURE
+        self.assertTrue(try_transition_node("node_under_test", trans_id, self.context))
+
+        trans_id = Transition.TRANSITION_ACTIVATE
+        self.assertTrue(try_transition_node("node_under_test", trans_id, self.context))
+
+        trans_id = Transition.TRANSITION_CONFIGURE
+        self.assertTrue(
+            try_transition_node("lanelet2_map_loader", trans_id, self.context)
+        )
+
+        trans_id = Transition.TRANSITION_ACTIVATE
+        self.assertTrue(
+            try_transition_node("lanelet2_map_loader", trans_id, self.context)
+        )
+
+        # Need to wait for roadway_object's WMListener to build it routing graph
+        # Note: This value was arbitrarily chosen. It could probably be reduced.
+        time.sleep(5)
+
+        external_object_list = make_external_object_list()
+        publish_msg(
+            self.test_harness_node,
+            self.test_harness_node.external_object_list_pub,
+            external_object_list,
+            self.context,
+        )
+
+        spin_node_until(
+            self.test_harness_node,
+            LenIncreases(self.test_harness_node.roadway_obstacle_lists),
```

Comment:
This is getting initialized. It will create a `LenIncreases` instance that gets passed into the function. In C++ terms, this would be a function object.

Reply:
Makes perfect sense, thanks